cylc/flow/network/schema.py

```diff
@@ -1182,4 +1183,104 @@ class Mutations(ObjectType):
         description=TaskActions._meta.description)
 
 
-schema = Schema(query=Queries, mutation=Mutations)
+# ** Subscription Related ** #
+
+def to_subscription(func: Callable, sleep_seconds: float = 5.) -> Callable:
+    """Wraps a function in a while-true-sleep, transforming
+    the function into an async-generator, used by the
+    websockets/subscriptions.
+
+    Args:
+        func (Callable): a callable.
+        sleep_seconds (float): asyncio sleep interval in seconds.
+    Returns:
+        Callable: a callable async-generator wrapping the original callable.
+    """
+    async def gen(*args: Any, **kwargs: Any) -> AsyncGenerator[Any, None]:
+        """
+        Args:
+            *args: Variable length argument list, varies as per schema.
+            **kwargs: Arbitrary keyword arguments, varies as per schema.
+        Returns:
+            AsyncGenerator[Any, None]: an async generator that will yield values from
+                resolvers.
+        """
+        while True:
+            yield await func(*args, **kwargs)
+            await asyncio.sleep(sleep_seconds)
+    return gen
```

Comment:
@dwsutherland a function that converts the existing resolver methods into async-generators, expected for the websockets to work.